pkg/controller/operconfig/migration.go

```
@@ -97,3 +112,325 @@ func convertEgressFirewallToEgressNetworkPolicy(ctx context.Context, client cnoc
+		return len(nns.Object["egressIPs"].([]interface{})) > 0
+	}
+	return false
+}
+
+func hostSubnetHasEgressIpConfigAutomatic(hsn uns.Unstructured) bool {
+	if hsn.Object["egressIPs"] != nil && hsn.Object["egressCIDRs"] != nil {
+		return len(hsn.Object["egressIPs"].([]interface{})) > 0 && len(hsn.Object["egressCIDRs"].([]interface{})) > 0
+	}
+	return false
+}
+
+func hostSubnetHasEgressIpConfigManual(hsn uns.Unstructured) bool {
+	if hsn.Object["egressIPs"] != nil && hsn.Object["egressCIDRs"] != nil {
+		return len(hsn.Object["egressIPs"].([]interface{})) > 0 && len(hsn.Object["egressCIDRs"].([]interface{})) <= 0
+	}
+	return false
+}
+
+func labelNodeAndRemoveHostSubnetConfig(ctx context.Context, client cnoclient.Client, hsn uns.Unstructured) error {
+	// label node as egressassignable
+	hostStr := fmt.Sprintf("%v", hsn.Object["host"])
+	nodeObj, err := client.Default().Kubernetes().CoreV1().Nodes().Get(ctx, hostStr, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	if _, ok := nodeObj.Labels["k8s.ovn.org/egress-assignable"]; !ok {
+		nodeObj.Labels["k8s.ovn.org/egress-assignable"] = ""
+		if err := apply.ApplyObject(ctx, client, nodeObj, ""); err != nil {
+			return err
+		}
+	}
+
+	// now remove egressIP config from hostsubnet
+	hsn.Object["egressCIDRs"] = nil
+	hsn.Object["egressIPs"] = nil
+	if err := apply.ApplyObject(ctx, client, &hsn, ""); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func deleteCloudPrivateIpConfigs(ctx context.Context, client cnoclient.Client, egressIps []interface{}) error {
+	for _, egressIp := range egressIps {
+		egressIpStr := fmt.Sprintf("%v", egressIp)
+		// check if cloudprivateipconfig exists before attempting to delete
+		if _, err := client.Default().Dynamic().Resource(gvrCloudPrivateIPConfig).Get(ctx, egressIpStr, metav1.GetOptions{}); err != nil {
+			errMessage := fmt.Sprintf("cloudprivateipconfigs.cloud.network.openshift.io \"%s\" not found", egressIpStr)
+			if err.Error() == errMessage {
```

Use `IsNotFound` from `k8s.io/apimachinery/pkg/api/errors`

It would make sense to directly `Delete` the object nad just ignore the error if `IsNotFound`, no need to lookup.